app/models/event_stream.rb

```diff
@@ -30,6 +30,10 @@ class EventStream < ApplicationRecord
   belongs_to :physical_chassis, :inverse_of => :event_streams
   belongs_to :physical_switch, :inverse_of => :event_streams
 
+  virtual_column :ext_management_system_name, :type => :string, :uses => :ext_management_system
+  virtual_column :host_name,                  :type => :string, :uses => :host
+  virtual_column :vm_or_template_name,        :type => :string, :uses => :vm_or_template
```

Comment:
great idea. lets use virtual delegates

```suggestion
  virtual_delegate :name, :to => :ext_management_system, :prefix => true, :allow_nil => true, :type => :string
  virtual_delegate :name, :to => :host, :prefix => true, :allow_nil => true, :type => :string
  virtual_delegate :name, :to => :vm_or_template, :prefix => true, :allow_nil => true, :type => :string
```